player/screenshot.c

```diff
@@ -454,6 +454,18 @@ void screenshot_to_file(struct MPContext *mpctx, const char *filename, int mode,
     ctx->osd = old_osd;
 }
 
+void screenshot_make(struct MPContext *mpctx, struct mp_image *img)
+{
+    screenshot_ctx *ctx = mpctx->screenshot_ctx;
+
+    struct image_writer_opts *opts = mpctx->opts->screenshot_image_opts;
+    char *filename = gen_fname(ctx, image_writer_file_ext(opts));
+    if (filename)
+        write_screenshot(mpctx, img, filename, NULL, true);
+    talloc_free(filename);
+    talloc_free(img);
+}
```

Comment:
I don't really like this approach, and also synchronization of any form is missing.